.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,11 +1,34 @@
+## Description
+<!--
+Describe your changes in detail.
+Write your text outside the comment blocks delimited by the "arrows".
+-->
+
+## How has this been tested?
+<!--
+Please describe in detail how you tested your changes.
+Include details of your testing environment, and the tests you ran
+to see how your change affects other areas of the code.
+You can upload screenshots/GIFs or link to videos as well.
+-->
+
+## Checklist
+<!--
+Go over all the following points, and put an `x` in all the boxes that apply.
+If you're unsure about any of these, don't hesitate to ask. We're here to help!
+-->
+
+- [ ] I have read the `CONTRIBUTING.md` document. <!-- Required. -->
+- [ ] I have updated the documentation accordingly. <!-- Required. -->
+- [ ] I have added unit tests to cover my changes. <!-- Recommended, but not required. -->
+
```

Comment:
This looks good if we want to make tests mandatory. Otherwise I think it could confuse contributors making them think that's the case. Maybe this should be clarified in the comment.

Reply:
Considering the vast majority of the codebase doesn't have unit tests yet, it seems a bit unfair to ask users to do this. Maybe we could add this checkbox to the template later, but I think for now we should only have the first two.

Reply:
> 
> 
> Considering the vast majority of the codebase doesn't have unit tests yet, it seems a bit unfair to ask users to do this. Maybe we could add this checkbox to the template later, but for now we would only have the first two.

But you've got to start adding Unit Tests somewhere, and lets be honest, no-ones going to be taking on writing large amounts of unit tests against the current codebase any time soon.

Reply:
I would prefer to keep this encouragement, especially now that https://github.com/godotengine/godot/pull/42430 has been merged.